### PR TITLE
IC2: Add converter for reactor components to get stored heat.

### PR DESCRIPTION
**Pull request by Engineer12798**
With coolant cells and other heat storing items, there is no uniform way in game to extract the heat stored in these items. To do so, would require parsing the NBT of the item stack, which is difficult enough as is, but would also require reading different NBT tags depending on the mod the item came from. To resolve this problem, this PR adds in a converter which will get the stored heat of any item implementing `IReactorComponent`.

The way the PR works may be jank and not work with some components. There may exist cases where the stored heat and max heat of a component are dependent on the reactor context, which this converter does not provide when trying to get the stored heat.

### CODE REVIEW

**Comment by OneEyeMaker** [comment]
Despite this implementation will work in certain circumstances, one doesn't solve mentioned issue properly. (Even more, one can cause an exception).

Instead I suggest to extend [`DriverReactor`](https://github.com/GTNewHorizons/OpenComputers/blob/master/src/main/scala/li/cil/oc/integration/ic2/DriverReactor.java) and [`DriverReactorChamber`](https://github.com/GTNewHorizons/OpenComputers/blob/master/src/main/scala/li/cil/oc/integration/ic2/DriverReactorChamber.java) classes and add necessary methods there.

And only then add your implementation for handling components outside reactor. Make sure to mark it *explicit* and *unsafe* so players understand what they do. Currently, from player's perspective, these properties you added look like regular (*safe*) ones.

**Review by OneEyeMaker — CHANGES_REQUESTED** [review]
Good job! Now this feature is almost complete.
Just find a way to tell players that original approach is not 100% reliable.

File: src/main/scala/li/cil/oc/integration/ic2/ConverterReactorHeatStorage.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    public void convert(final Object value, final Map<Object, Object> output) {
+        if (value instanceof ItemStack) {
+            final ItemStack stack = (ItemStack) value;
+            final Item item = stack.getItem();
+            if (item instanceof IReactorComponent) {
+                final IReactorComponent reactorComponent = (IReactorComponent) item;
+                int maxHeat = reactorComponent.getMaxHeat(null, stack, 0, 0);
+                if (maxHeat > 0) {
+                	output.put("heat", reactorComponent.getCurrentHeat(null, stack, 0, 0));
+                	output.put("maxHeat", maxHeat);
```
[the remark this answers is not in the document]
Comment (Ktmi) [reply src/main/scala/li/cil/oc/integration/ic2/ConverterReactorHeatStorage.java:20]: I'm just going to drop the converter for the item. This is going to be used primarily in the context of reactors anyway, so it shouldn't be too big a deal to to not have the item coverter. And even if user's didn't use the unsafe variable, the converter would still run whenever getting the item stack, regardless of reading the variable or not.

[the remark this answers is not in the document]
Comment (Engineer12798) [reply src/main/scala/li/cil/oc/integration/ic2/ConverterReactorHeatStorage.java]: Got rid of the converter. This is going to be primarily used with reactors anyway.

**Review by OneEyeMaker — APPROVED** [review]

**Pull request merged**